analyzers/tests/SonarAnalyzer.UnitTest/Helpers/NavigationHelperTest.cs

```diff
@@ -145,5 +173,18 @@ public void GetPrecedingStatement()
             statements[1].GetPrecedingStatement().Should().BeEquivalentTo(statements[0]);
             statements[0].GetPrecedingStatement().Should().Be(null);
         }
+
+        [TestMethod]
+        public void GetPrecedingStatementTopLevelStatements()
+        {
+            var globalStatements = syntaxTreeTopLevelStatement.GetRoot()
+                                                              .ChildNodes()
+                                                              .Select(x => x.ChildNodes().FirstOrDefault())
+                                                              .OfType<StatementSyntax>()
+                                                              .ToArray();
+
+            globalStatements[1].GetPrecedingStatement().Should().BeEquivalentTo(globalStatements[0]);
+            globalStatements[0].GetPrecedingStatement().Should().Be(null);
+        }
```

Comment:
I would rewrite the test to avoid running a similar alg in the test with the one in production code. We can look for the statements and check them directly. 

WDYT about the following snippet?
```suggestion
        [TestMethod]
        public void GetPrecedingStatementTopLevelStatements()
        {
            var variableDeclarators = syntaxTreeTopLevelStatement.GetRoot().DescendantNodes().OfType<LocalDeclarationStatementSyntax>().ToArray();
            var aDeclaration = variableDeclarators[0];
            var bDeclaration = variableDeclarators[0];

            aDeclaration.GetPrecedingStatement().Should().Be(null);
            bDeclaration.GetPrecedingStatement().Should().BeEquivalentTo(aDeclaration);
        }
```